mocpy/tmoc.py

```diff
@@ -0,0 +1,259 @@
+        A value near 0 shows a sparse temporal moc (i.e. the moc does not cover a lot
+        of time and covers very distant times. A value near 1 means that the moc covers
+        a lot of time without big pauses.
+        """
+        return self.total_duration / float(self.max_time - self.min_time)
+
+    @property
+    def min_time(self):
+        """
+        Return the min time of the tmoc in jd
+        """
+        return self._interval_set.min / TimeMoc.DAY_MICRO_SEC
+
+    @property
+    def max_time(self):
+        """
+        Return the max time of the tmoc in jd
+        """
+        return self._interval_set.max / TimeMoc.DAY_MICRO_SEC
+
+    def filter_table(self, table, t_column, format='decimalyear', keep_inside=True):
+        """
+        Filter an astropy.table.Table to keep only rows inside (or outside) the tmoc instance
+
+        :param table: the full astropy table to filter
+        :param t_column: the name of the time column to consider for filtering the table
+        :param format: the format of the time column
+        :param keep_inside: boolean describing if we want to keep rows inside or outside the tmoc
+        :return: the filtered Table (astropy.table)
+        """
+
+        time_arr = Time(table[t_column], format=format, scale='tai')
+        pix_arr = (time_arr.jd * TimeMoc.DAY_MICRO_SEC)
+        pix_arr = pix_arr.astype(int)
+
+        intervals_arr = np.array(self._interval_set.intervals)
+        inf_arr = np.vstack([pix_arr[i] >= intervals_arr[:, 0] for i in range(pix_arr.shape[0])])
+        sup_arr = np.vstack([pix_arr[i] <= intervals_arr[:, 1] for i in range(pix_arr.shape[0])])
+
+        if keep_inside:
+            res = inf_arr & sup_arr
+            filtered_rows = np.any(res, axis=1)
+        else:
+            res = ~inf_arr | ~sup_arr
+            filtered_rows = np.all(res, axis=1)
+
+        return table[filtered_rows]
+
+    def add_fits_header(self, tbhdu):
+        """
+        Add TIMESYS to the header of the fits describing the tmoc
+
+        :param tbhdu: fits python object
+        """
+        tbhdu.header['TIMESYS'] = ('JD', 'ref system JD BARYCENTRIC TT, 1 microsec level 29')
+
+    def plot(self, title='TimeMoc', view=(None, None)):
+        """
+        Plot the tmoc with matplotlib
+
+        :param title: the title of the plot
+        :param view: a tuple (x_min, x_max) defining the time window of the plot
+        """
+        assert not self._interval_set.empty(), ValueError('Empty time moc instance')
+
+        plot_order = 15
+        if self.max_order > plot_order:
+            plotted_moc = self.degrade_to_order(plot_order)
+        else:
+            plotted_moc = self
+
+        plotted_moc._interval_set.intervals
+
+        min_jd, max_jd = view
+
+        if not min_jd:
+            min_jd = plotted_moc.min_time
+        if not max_jd:
+            max_jd = plotted_moc.max_time
+
+        assert max_jd > min_jd, ValueError('max_jd must be > to min_jd')
```

Comment:
Same here: suggest to use if and raise.
This assert, ValueError probably doesn't do what you want it to in some cases, or at the very least is non-standard in Python.

Reply:
I was just suggesting to not use `assert`, but do input validation like this instead:
```
if max_jd <= min_jd:
    raise ValueError('Invalid selection: max_jd = {} must be > to min_jd = {}'.format(max_id, min_id)
```